server/conn_stmt.go

```diff
@@ -98,8 +98,10 @@ func (cc *clientConn) handleStmtPrepare(ctx context.Context, sql string) error {
 			}
 		}
 
-		if err := cc.writeEOF(0); err != nil {
-			return err
+		if !(cc.capability&mysql.ClientDeprecateEOF > 0) {
```

Comment:
Why not `cc.capability&mysql.ClientDeprecateEOF  == 0`?

Reply:
That would be ok as well.
```suggestion
		if !(cc.capability&mysql.ClientDeprecateEOF == 0) {
```

Reply:
Done in 0abd5e3367a1a0caf6258e1d91eca020e4bdaf89